web/web.go

```diff
@@ -503,9 +508,16 @@ func serveDebug(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// Ready sets Handler to be ready.
-func (h *Handler) Ready() {
-	h.ready.Store(1)
+// SetReady sets the ready status of our web Handler
+func (h *Handler) SetReady(v bool) {
+	if v {
+		h.ready.Store(1)
```

Comment:
I was considering using existing `ready` variable and adding a collector that exposes it as metric, but then the logic in our handler is tied to a metric value, which doesn't seem right to me, plus it would be a bit more code, so in the end decided this is probably more straightforward.

Reply:
`GaugeFunc` would simplify code but it needs to access `Handler.ready` so I would have to either:
- move `ready` from `Handler` into `metrics` struct
- pass `ready` pointer into metrics
- have `prometheus_ready` metric defined on the `Handler` itself, rather than `metrics`

is any of that considered better than the current PR?

Reply:
Current approach is fine then